src/battle_util.c

```diff
@@ -10930,10 +10925,25 @@ bool8 AreBattlersOfOppositeGender(u8 battler1, u8 battler2)
     return (gender1 != MON_GENDERLESS && gender2 != MON_GENDERLESS && gender1 != gender2);
 }
 
-u32 CalcSecondaryEffectChance(u8 battlerId, u8 secondaryEffectChance)
+u32 CalcSecondaryEffectChance(u8 battlerId, u16 move, u8 *rng)
 {
+    u32 baseChance = gBattleMoves[move].secondaryEffectChance;
+
+    // In cases where we have a move with TWO move effects,
+    // calculate its RNG for each effect separately
+    // If the second move effect has an additional % chance,
+    // make sure to use that instead. Only works with moves
+    // with FLAG_ADDITIONAL_ARG_MOVE_EFFECT
+    if (gBattleMoves[move].flags & FLAG_ADDITIONAL_ARG_MOVE_EFFECT
+        && CURRENT_MOVE_EFFECT == ARG_MOVE_EFFECT_ONLY(move))
+    {
+        (*rng)++;
```

Comment:
Good call on using `RNG_SECONDARY_EFFECT2` instead of rng++. I wasn't sure where to put the check for rng so I threw it in there. It's entangled with the move effect so it seemed the sensible choice.